src/main/kotlin/com/cognifide/gradle/aem/common/build/BuildScope.kt

```diff
@@ -7,21 +7,31 @@ class BuildScope {
     private val cache = mutableMapOf<String, Any>()
 
     @Suppress("unchecked_cast")
-    fun <T : Any> get(key: String): T? {
-        return cache[key] as T
-    }
-
-    fun <T : Any> getOrPut(key: String, defaultValue: () -> T): T = getOrPut(key, defaultValue, false)
+    fun <T : Any> get(key: String): T? = cache[key] as T?
 
     @Suppress("unchecked_cast")
+    fun <T : Any> getOrPut(key: String, defaultValue: () -> T) = cache.getOrPut(key, defaultValue) as T
+
     fun <T : Any> getOrPut(key: String, defaultValue: () -> T, invalidate: Boolean): T {
         return if (invalidate) {
             val value = defaultValue()
             put(key, value)
             value
         } else {
-            cache.getOrPut(key, defaultValue) as T
+            getOrPut(key, defaultValue)
+        }
+    }
+
+    fun <T : Any> tryGetOrPut(key: String, defaultValue: () -> T?): T? {
+        var result: T? = get(key)
+        if (result == null) {
```

Comment:
`result = result :? defaultValue()`

Reply:
nope, in case of not null value cache must be filled (so that more than one operation need to be done, not only assignment)